### DISCUSSION TITLE
Why sunset - sunrise != daylight duration

### DISCUSSION THREAD

**Discussion by danielwerg** [open-meteo/open-meteo#1104] [Q&A] [answered]
I'm trying to calculate remaining daylight and noticed this.

Getting data for London today:

sunrise: `2024-11-08T07:07`
sunset: `2024-11-08T16:21`
daylight_duration: `33222.79`
sunrise tomorrow: `2024-11-09T07:09`
now: `2024-11-08T07:00`

```bash
echo "$(date -d "2024-11-08T16:21" +%s) - $(date -d "2024-11-08T07:07" +%s)" | bc
```

```
33240
```

^ Slightly off from `daylight_duration`, in this case just around a minute.

Is there a reason for this?

**Comment by patrick-zippenfenig** [comment, marked as answer]
Hi,
sunrise/set are only given to the nearest minute in the ISO8601 format. The underlaying calculation is based on seconds as given in daylight duration. The difference of 1 minute is just rounding.
